# BUSINESS UNDERSTANDING
This project aims to help an aviation insurer reduce financial risk by identifying airplane makes and models with the lowest rates of total destruction and passenger injury. Using professional accident data from 1983 to 2023, we are evaluating modern, active aircraft to provide separate safety recommendations for small planes (20 or fewer passengers) and large commercial jets. By analyzing these trends and identifying external risk factors like weather or engine types, we will provide the client with a statistically backed "safe-to-insure" list for both market segments.


# Data understanding
Here is where we load the data and check missing values, we also check missing columns, import libraries,check the data types 

### STEP2:Data Loading and Inspection
Load in data from the relevant directory and inspect the dataframe.

1. Inspect NaNs and datatypes
2. Print out summary statistics

In [1]:
#importing libraries
import pandas as pd 
import numpy as np 

In [2]:
#loading the dataset
aviation_df= pd.read_csv("AviationData.csv", encoding='latin1')

C:\Users\User\AppData\Local\Temp\ipykernel_6976\1745053355.py:2: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  aviation_df= pd.read_csv("AviationData.csv", encoding='latin1')


In [3]:
#checking the dataset
aviation_df

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
0,20001218X45444,Accident,SEA87LA080,1948-10-24,"MOOSE CREEK, ID",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,UNK,Cruise,Probable Cause,NaN
1,20001218X45447,Accident,LAX94LA336,1962-07-19,"BRIDGEPORT, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,4.0,0.0,0.0,0.0,UNK,Unknown,Probable Cause,19-09-1996
2,20061025X01555,Accident,NYC07LA005,1974-08-30,"Saltville, VA",United States,36.922223,-81.878056,NaN,NaN,...,Personal,NaN,3.0,NaN,NaN,NaN,IMC,Cruise,Probable Cause,26-02-2007
3,20001218X45448,Accident,LAX96LA321,1977-06-19,"EUREKA, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,IMC,Cruise,Probable Cause,12-09-2000
4,20041105X01764,Accident,CHI79FA064,1979-08-02,"Canton, OH",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,2.0,NaN,0.0,VMC,Approach,Probable Cause,16-04-1980
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88884,20221227106491,Accident,ERA23LA093,2022-12-26,"Annapolis, MD",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,0.0,1.0,0.0,0.0,NaN,NaN,NaN,29-12-2022
88885,20221227106494,Accident,ERA23LA095,2022-12-26,"Hampton, NH",United States,NaN,NaN,NaN,NaN,...,NaN,NaN,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN
88886,20221227106497,Accident,WPR23LA075,2022-12-26,"Payson, AZ",United States,341525N,1112021W,PAN,PAYSON,...,Personal,NaN,0.0,0.0,0.0,1.0,VMC,NaN,NaN,27-12-2022
88887,20221227106498,Accident,WPR23LA076,2022-12-26,"Morgan, UT",United States,NaN,NaN,NaN,NaN,...,Personal,MC CESSNA 210N LLC,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN


In [4]:
#load data set
states_df=pd.read_csv("USState_Codes.csv")

Observation pandas does not recoqnixe the (utf-8) hence why we used the encoding="latin1"

In [5]:
#checking dimensions
aviation_df.shape

(88889, 31)

In [6]:
print(f"The aviation dataframe has {aviation_df.shape[0]} rows and {aviation_df.shape[1]} columns")

The aviation dataframe has 88889 rows and 31 columns


Observation: The aviation_df has 88889 rows and 31 columns

In [7]:
#checking the columns
#see the exact names of each column
aviation_df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [8]:
#checking the last 5
aviation_df.tail()

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
88884,20221227106491,Accident,ERA23LA093,2022-12-26,"Annapolis, MD",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,0.0,1.0,0.0,0.0,NaN,NaN,NaN,29-12-2022
88885,20221227106494,Accident,ERA23LA095,2022-12-26,"Hampton, NH",United States,NaN,NaN,NaN,NaN,...,NaN,NaN,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN
88886,20221227106497,Accident,WPR23LA075,2022-12-26,"Payson, AZ",United States,341525N,1112021W,PAN,PAYSON,...,Personal,NaN,0.0,0.0,0.0,1.0,VMC,NaN,NaN,27-12-2022
88887,20221227106498,Accident,WPR23LA076,2022-12-26,"Morgan, UT",United States,NaN,NaN,NaN,NaN,...,Personal,MC CESSNA 210N LLC,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN
88888,20221230106513,Accident,ERA23LA097,2022-12-29,"Athens, GA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,0.0,1.0,0.0,1.0,NaN,NaN,NaN,30-12-2022


In [9]:
#checking statistics
aviation_df.describe().T

,count,mean,std,min,25%,50%,75%,max
Number.of.Engines,82805.0,1.146585,0.446510,0.0,1.0,1.0,1.0,8.0
Total.Fatal.Injuries,77488.0,0.647855,5.485960,0.0,0.0,0.0,0.0,349.0
Total.Serious.Injuries,76379.0,0.279881,1.544084,0.0,0.0,0.0,0.0,161.0
Total.Minor.Injuries,76956.0,0.357061,2.235625,0.0,0.0,0.0,0.0,380.0
Total.Uninjured,82977.0,5.325440,27.913634,0.0,0.0,1.0,2.0,699.0


In [10]:
#cheaking info
#check if the numbers are actual numbers of the computer is reading them as text
aviation_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

In [11]:
#checking null values with isna method
aviation_df.isna().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38757
Airport.Name              36185
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Registration.Number        1382
Make                         63
Model                        92
Amateur.Built               102
Number.of.Engines          6084
Engine.Type                7096
FAR.Description           56866
Schedule                  76307
Purpose.of.flight          6192
Air.carrier               72241
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
Report.Status              6384
Publication.Date          13771
dtype: i

In [12]:
#check unique values
for coln in aviation_df.columns:
    uni_value=aviation_df[coln].unique()
    print(f"{coln}\n {uni_value}\n\n")

Event.Id
 ['20001218X45444' '20001218X45447' '20061025X01555' ... '20221227106497'
 '20221227106498' '20221230106513']


Investigation.Type
 ['Accident' 'Incident']


Accident.Number
 ['SEA87LA080' 'LAX94LA336' 'NYC07LA005' ... 'WPR23LA075' 'WPR23LA076'
 'ERA23LA097']


Event.Date
 ['1948-10-24' '1962-07-19' '1974-08-30' ... '2022-12-22' '2022-12-26'
 '2022-12-29']


Location
 ['MOOSE CREEK, ID' 'BRIDGEPORT, CA' 'Saltville, VA' ... 'San Manual, AZ'
 'Auburn Hills, MI' 'Brasnorte, ']


Country
 ['United States' nan 'GULF OF MEXICO' 'Puerto Rico' 'ATLANTIC OCEAN'
 'HIGH ISLAND' 'Bahamas' 'MISSING' 'Pakistan' 'Angola' 'Germany'
 'Korea, Republic Of' 'Martinique' 'American Samoa' 'PACIFIC OCEAN'
 'Canada' 'Bolivia' 'Mexico' 'Dominica' 'Netherlands Antilles' 'Iceland'
 'Greece' 'Guam' 'Australia' 'CARIBBEAN SEA' 'West Indies' 'Japan'
 'Philippines' 'Venezuela' 'Bermuda' 'San Juan Islands' 'Colombia'
 'El Salvador' 'United Kingdom' 'British Virgin Islands' 'Netherlands'
 'Costa Rica' 'Mozamb

In [13]:
#checking if the aircrafts of the same types are given diffrent names
aviation_df['Make'].head(50)

0               Stinson
1                 Piper
2                Cessna
3              Rockwell
4                Cessna
5     Mcdonnell Douglas
6                Cessna
7                Cessna
8                Cessna
9        North American
10                Piper
11                Beech
12             Bellanca
13               Cessna
14               Navion
15                Beech
16              Enstrom
17               Cessna
18               Cessna
19                Smith
20               Cessna
21               Cessna
22                 Bell
23                Piper
24              Grumman
25               Cessna
26           Beechcraft
27                Maule
28               Cessna
29             Bellanca
30               Cessna
31          Air Tractor
32                Piper
33                Piper
34               Cessna
35                Piper
36               Cessna
37               Navion
38                Piper
39                Piper
40             Rockwell
41              

In [14]:
#check all columns
print(aviation_df.columns.tolist())

['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date', 'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code', 'Airport.Name', 'Injury.Severity', 'Aircraft.damage', 'Aircraft.Category', 'Registration.Number', 'Make', 'Model', 'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description', 'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries', 'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured', 'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status', 'Publication.Date']


In [15]:
#make a copy
df=aviation_df.copy()

Observation:The data set has alot of null values and missing values and alot of unnecessary columns that need to be dropped and needs proper cleaning since it is very dirty 

# DATA PREPARATION 


In [16]:
#convert columns to lower case,remove white space,strip,change naming
df.columns=df.columns.str.replace(" ",".",regex=True).str.lower().str.strip()

In [17]:
df

,event.id,investigation.type,accident.number,event.date,location,country,latitude,longitude,airport.code,airport.name,...,purpose.of.flight,air.carrier,total.fatal.injuries,total.serious.injuries,total.minor.injuries,total.uninjured,weather.condition,broad.phase.of.flight,report.status,publication.date
0,20001218X45444,Accident,SEA87LA080,1948-10-24,"MOOSE CREEK, ID",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,UNK,Cruise,Probable Cause,NaN
1,20001218X45447,Accident,LAX94LA336,1962-07-19,"BRIDGEPORT, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,4.0,0.0,0.0,0.0,UNK,Unknown,Probable Cause,19-09-1996
2,20061025X01555,Accident,NYC07LA005,1974-08-30,"Saltville, VA",United States,36.922223,-81.878056,NaN,NaN,...,Personal,NaN,3.0,NaN,NaN,NaN,IMC,Cruise,Probable Cause,26-02-2007
3,20001218X45448,Accident,LAX96LA321,1977-06-19,"EUREKA, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,IMC,Cruise,Probable Cause,12-09-2000
4,20041105X01764,Accident,CHI79FA064,1979-08-02,"Canton, OH",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,2.0,NaN,0.0,VMC,Approach,Probable Cause,16-04-1980
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88884,20221227106491,Accident,ERA23LA093,2022-12-26,"Annapolis, MD",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,0.0,1.0,0.0,0.0,NaN,NaN,NaN,29-12-2022
88885,20221227106494,Accident,ERA23LA095,2022-12-26,"Hampton, NH",United States,NaN,NaN,NaN,NaN,...,NaN,NaN,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN
88886,20221227106497,Accident,WPR23LA075,2022-12-26,"Payson, AZ",United States,341525N,1112021W,PAN,PAYSON,...,Personal,NaN,0.0,0.0,0.0,1.0,VMC,NaN,NaN,27-12-2022
88887,20221227106498,Accident,WPR23LA076,2022-12-26,"Morgan, UT",United States,NaN,NaN,NaN,NaN,...,Personal,MC CESSNA 210N LLC,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN


### STEP3:Data cleaning
Filtering out data
Main focus:
1. aircraft.category
2. amateur.built
3. event.date

##### aircraft.category

In [18]:
#cheaking the aircraft category
df["aircraft.category"].unique()

array([nan, 'Airplane', 'Helicopter', 'Glider', 'Balloon', 'Gyrocraft',
       'Ultralight', 'Unknown', 'Blimp', 'Powered-Lift', 'Weight-Shift',
       'Powered Parachute', 'Rocket', 'WSFT', 'UNK', 'ULTR'], dtype=object)

So in aircraft.category the client is only intrested in the ariplane so we filter out all other information

In [19]:
#filtering out to obtain only airplane
df= df[df["aircraft.category"] == "Airplane"]

In [20]:
df["aircraft.category"].unique()

array(['Airplane'], dtype=object)

So we have succesfully filtered out all the other unneccesary data in aircraft.category to obtain airplanes only

#### amateur.build

In [21]:
df["amateur.built"].unique()

array(['No', 'Yes', nan], dtype=object)

In [22]:
#filtering out the amateur.built into professional builds only
df= df[df["amateur.built"] == "No"]

In [23]:
#checking for succesfull filtering
df["amateur.built"].unique()

array(['No'], dtype=object)

amateur.built has succesfully been filtered to only  professional builds only

##### event.date

In [24]:
#convert date to datetime object
df['event.date'] = pd.to_datetime(df['event.date'], errors='coerce')

In [25]:
#fitering event.date into data that is 40 year younger
df=df[df["event.date"].dt.year >= 1983]

In [26]:
#checking if the event.date has been filtered
df["event.date"].unique()

<DatetimeArray>
['1983-03-18 00:00:00', '1983-03-20 00:00:00', '1983-04-02 00:00:00',
 '1983-08-21 00:00:00', '1983-10-28 00:00:00', '1983-11-08 00:00:00',
 '1983-11-13 00:00:00', '1983-12-22 00:00:00', '1984-03-14 00:00:00',
 '1984-06-11 00:00:00',
 ...
 '2022-12-05 00:00:00', '2022-12-08 00:00:00', '2022-12-09 00:00:00',
 '2022-12-10 00:00:00', '2022-12-12 00:00:00', '2022-12-13 00:00:00',
 '2022-12-14 00:00:00', '2022-12-15 00:00:00', '2022-12-16 00:00:00',
 '2022-12-26 00:00:00']
Length: 6769, dtype: datetime64[ns]

Observation all the dataset has succesfully been filtered to the clients requirments

### STEP4:Cleaning and Constructing Key Measurables

Main focus:
1. Construct metric for fatal/serious injuries.
2. Aircraft.Damage

#####  Construct metric for fatal/serious injuries.

In [27]:
# Impute missing injury and passenger counts with 0
injury_cols = ['total.fatal.injuries', 'total.serious.injuries', 'total.minor.injuries', 'total.uninjured']
df[injury_cols] = df[injury_cols].fillna(0)

In [28]:
# Estimate total passengers
df["total_passengers"] = df[injury_cols].sum(axis=1)

In [29]:
# Construct metric for fatal/serious injuries fraction
df["fatal_serious_fraction"] = np.where(
    df["total_passengers"] > 0,
    (df["total.fatal.injuries"] + df["total.serious.injuries"]) / df["total_passengers"],
    0
)

In [30]:
# Clean aircraft.damage column and fill it with unknown
df['aircraft.damage'] = df['aircraft.damage'].fillna("Unknown")


In [31]:
#create a destroyed derived column (1 if destroyed, 0 otherwise)
df["is_destroyed"] = np.where(df['aircraft.damage'] == "Destroyed", 1, 0)

In [32]:
#checking the aircraft.damage column with is_destroyed column
df[['aircraft.damage','is_destroyed']]

,aircraft.damage,is_destroyed
4149,Minor,0
4150,Minor,0
4171,Destroyed,1
4285,Unknown,0
5957,Minor,0
...,...,...
88869,Substantial,0
88873,Substantial,0
88876,Substantial,0
88877,Substantial,0


### STEP5 Investigate the Make Column
1. Identify cleaning tasks here.
2. List cleaning tasks clearly in markdown.
3. Execute the cleaning tasks.
4. For your analysis, keep Makes with a reasonable number (put the threshold at 50).

### Here we do the following
1. Standerdize and make them all lower case
2. Replace whitespace with (.)
3. Ensure common brands are unified
4. Filtering data for only the significant makes

In [33]:
# Clean Make column
df['make'] = df['make'].str.lower().str.strip().str.replace(" ",".",regex=True)


In [34]:
# 2. Identify makes with 50 or more incidents
make_counts = df['make'].value_counts()
makes_to_keep = make_counts[make_counts >= 50].index

In [35]:
print(df.count())

event.id                  21447
investigation.type        21447
accident.number           21447
event.date                21447
location                  21441
country                   21446
latitude                  19169
longitude                 19163
airport.code              13983
airport.name              14070
injury.severity           20634
aircraft.damage           21447
aircraft.category         21447
registration.number       21241
make                      21444
model                     21429
amateur.built             21447
number.of.engines         18896
engine.type               17491
far.description           20967
schedule                   2521
purpose.of.flight         17767
air.carrier               10162
total.fatal.injuries      21447
total.serious.injuries    21447
total.minor.injuries      21447
total.uninjured           21447
weather.condition         18468
broad.phase.of.flight      2828
report.status             16784
publication.date          20490
total_pa

### STEP6:Inspect Model Column
1. Get rid of any NaNs.
2. Inspect the column and counts for each model/make.
3. Consider whether model labels are unique to each make. If not, create a derived column that is a unique identifier for a given plane type

In [36]:
#drop rows where 'model' is missing (NaN)
df= df.dropna(subset=['model'])

In [37]:
#clean the Model names (lowercase and strip spaces)
df['model'] = df['model'].str.lower().str.strip().str.replace(" ",".",regex=True)

In [38]:
#create a unique identifier: 'Make_Model'
df['make_model'] = df['make'] + "_" + df['model']

In [39]:
#inspect the results
print(f"Total unique plane types: {df['make_model'].nunique()}")
print(df['make_model'].value_counts().head(10))

Total unique plane types: 4355
make_model
cessna_172     769
boeing_737     403
cessna_152     316
cessna_182     304
cessna_172s    276
piper_pa28     273
cessna_172n    249
cessna_180     213
cessna_172m    180
cessna_150     179
Name: count, dtype: int64


### STEP7:Cleaning Other Columns
There are other columns containing data that might be related to the outcome of an accident. We list a few here:

1. Engine.Type
2. Weather.Condition
3. Number.of.Engines
4. Purpose.of.flight
5. Broad.phase.of.flight
6. Inspect and identify potential cleaning tasks in each of the columns. Execute those cleaning tasks.



In [40]:
#clean the categorical text columns
text_cols = ['engine.type', 'weather.condition', 'purpose.of.flight', 'broad.phase.of.flight']

for col in text_cols:
    #convert to Title Case (e.g., 'vMC' becomes 'Vmc') and remove spaces
    df[col] = df[col].str.title().str.strip()
    
    #replace 'Unk' or empty strings with 'Unknown'
    df[col] = df[col].replace(['Unk', '', 'None'], 'Unknown')

In [41]:
#fix the 'Number.of.Engines' column
#some rows might have 0.0 or 10.0 (unlikely for standard insurance models)
#we force it to be numeric and cap it at a reasonable range
df['number.of.engines'] = pd.to_numeric(df['number.of.engines'], errors='coerce')
#replace 0 engines with NaN because an airplane must have at least 1 engine to be in this study
df.loc[df['number.of.engines'] < 1, 'number.of.engines'] = np.nan

In [42]:
#inspection to see the most common values now
for col in text_cols:
    print(f"Top 5 values in {col}")
    print(df[col].value_counts().head(5))
    print("\n")

Top 5 values in engine.type
engine.type
Reciprocating    15014
Turbo Prop        1251
Turbo Fan          926
Unknown            134
Turbo Jet          133
Name: count, dtype: int64


Top 5 values in weather.condition
weather.condition
Vmc        17082
Imc         1068
Unknown      309
Name: count, dtype: int64


Top 5 values in purpose.of.flight
purpose.of.flight
Personal              11713
Instructional          2753
Aerial Application      933
Business                498
Unknown                 374
Name: count, dtype: int64


Top 5 values in broad.phase.of.flight
broad.phase.of.flight
Landing        1255
Takeoff         509
Cruise          264
Approach        234
Maneuvering     167
Name: count, dtype: int64




### STEP8:Column Removal
Inspect the dataframe and drop any columns that have too many NaNs.

For this exercise, keep all columns with more than 20,000 non-nulls.

In [43]:
#check how many non-null values are in each column
print("Non-null counts before dropping:")
print(df.count())

Non-null counts before dropping:
event.id                  21429
investigation.type        21429
accident.number           21429
event.date                21429
location                  21423
country                   21428
latitude                  19160
longitude                 19154
airport.code              13975
airport.name              14060
injury.severity           20618
aircraft.damage           21429
aircraft.category         21429
registration.number       21225
make                      21428
model                     21429
amateur.built             21429
number.of.engines         18873
engine.type               17486
far.description           20949
schedule                   2514
purpose.of.flight         17758
air.carrier               10153
total.fatal.injuries      21429
total.serious.injuries    21429
total.minor.injuries      21429
total.uninjured           21429
weather.condition         18459
broad.phase.of.flight      2822
report.status             16775
publica

In [44]:
#droping using dropna method
#dropping columns that did not make the 20,000 threshold 
df = df.dropna(thresh=20000, axis=1)

In [45]:
#checking the columns that made it more than 20,000
print(df.count())

event.id                  21429
investigation.type        21429
accident.number           21429
event.date                21429
location                  21423
country                   21428
injury.severity           20618
aircraft.damage           21429
aircraft.category         21429
registration.number       21225
make                      21428
model                     21429
amateur.built             21429
far.description           20949
total.fatal.injuries      21429
total.serious.injuries    21429
total.minor.injuries      21429
total.uninjured           21429
publication.date          20474
total_passengers          21429
fatal_serious_fraction    21429
is_destroyed              21429
make_model                21428
dtype: int64


In [46]:
#to see which columns survived the cut
print(df.columns.tolist())

['event.id', 'investigation.type', 'accident.number', 'event.date', 'location', 'country', 'injury.severity', 'aircraft.damage', 'aircraft.category', 'registration.number', 'make', 'model', 'amateur.built', 'far.description', 'total.fatal.injuries', 'total.serious.injuries', 'total.minor.injuries', 'total.uninjured', 'publication.date', 'total_passengers', 'fatal_serious_fraction', 'is_destroyed', 'make_model']


In [47]:
#check on the dataframe shape
print(f"New dataframe shape: {df.shape}")

New dataframe shape: (21429, 23)


Observation: Only 23 columns made the cut for the 20,000 threshold

In [48]:
#Saving the cleaned dataset
df.to_csv("cleaned_accidents_data.csv",index=None)